roles/yum_repos/tasks/main.yml

```diff
@@ -1,16 +1,25 @@
 ---
-- name: check if yum is present
-  command: python -c "import yum"
-  register: yum_installed
-  ignore_errors: yes
+- name: collect yum repos
+  yum:
+    list: repos
+  register: yum_repos
+  when: ansible_pkg_mgr == 'yum'
+
+- name: storing the yum repos data
+  set_fact:
+    stockpile_yum_repos: "{{ yum_repos.results }}"
+  when: yum_repos is not skipped
+
+- name: collect dnf repos
+  dnf:
+    list: repos
+  register: dnf_repos
+  when: ansible_pkg_mgr == 'dnf'
+
+- name: storing the dnf repos data
+  set_fact:
+    stockpile_yum_repos: "{{ yum_repos.results }}"
```

Comment:
needs to be `stockpile_dnf_repos` and `{{ dnf_repos.results }}` but in theory idea is nice :)

Reply:
Fixed.

Reply:
stockpile_dnf_repos: "{{ yum_repos.results }}" needs to be  stockpile_dnf_repos: "{{ dnf_repos.results }}"